File: 4.ROS/Tasks/Task_1/count_ws/src/pubCount_pkg/src/pub.cpp
#include <ros/ros.h>
#include <std_msgs/Int32.h>

int main(int argc, char** argv)  // argc & argv to make node conect each other in ROS system
{
    // crate node
    ros::init(argc, argv, "Publisher");
    ros::NodeHandle nh; 
    ros::Rate rate(1); // publish rate = 1 hz
    // create publisher object
    ros::Publisher pub;
    // publisher object publisher via counter topic 
    pub = nh.advertise<std_msgs::Int32>("counter", 1);

    std_msgs::Int32 count;
    count.data = 100; // create count msg

    while (ros::ok())  // loop until ROS shutdowm
    {
        pub.publish(count);  // publish msg
        ros::spinOnce(); //
        rate.sleep();  // to make sure that we publish with 1 hz
        ++count.data; 
    } 

    return 0;
}

File: 4.ROS/Tasks/Task_1/count_ws/src/pubCount_pkg/scripts/pub.py
#!/usr/bin/python3

import rospy
from std_msgs.msg import Int32

def talker():
    # create node
    rospy.init_node("Talker") 
    # create topic
    pub = rospy.Publisher("count_topic", Int32, queue_size=1)
    rate = rospy.Rate(1)  # publish rate = 1 hz

    count_msg = 100
    while not rospy.is_shutdown():  # untill ROS shutdown
        count_msg += 1
        rospy.loginfo(count_msg)   # print count_msg on screen
        pub.publish(count_msg)     # publish count_msg on /count_topic
        rate.sleep()

if __name__ == "__main__":
    try:
        talker()
    except rospy.ROSInterruptException:
        pass